tests/ballerina-test/src/test/resources/test-src/expressions/lambda/iterable/basic-iterable.bal

```diff
@@ -111,6 +111,91 @@ function testBasicMap2 () (string[] values) {
     return;
 }
 
+struct Employee {
+    int id;
+    string name;
+    float salary;
+}
+
+struct EmployeeCompatible {
+    int id;
+    string name;
+    float salary;
+}
+
+struct EmployeeSalary {
+    int id;
+    float salary;
+}
+
+struct EmployeeSalaryCompatible {
+    int id;
+    float salary;
+}
+
+function getSalary(Employee e) (EmployeeSalary) {
+    EmployeeSalary s = {id: e.id, salary: e.salary};
+    return s;
+}
+
+function getSalaryCompatibleInput(EmployeeCompatible e) (EmployeeSalary) {
+    EmployeeSalary s = {id: e.id, salary: e.salary};
+    return s;
+}
+
+function getSalaryCompatibleOutput(Employee e) (EmployeeSalaryCompatible ) {
+    EmployeeSalaryCompatible s = {id: e.id, salary: e.salary};
+    return s;
+}
+
+function getSalaryCompatibleInputOutput(EmployeeCompatible e) (EmployeeSalaryCompatible) {
+    EmployeeSalaryCompatible s = {id: e.id, salary: e.salary};
+    return s;
+}
+
+function createTable() (table<Employee>) {
+    table<Employee> dt = {};
+
+    Employee e1 = {id:1, name:"A", salary:100};
+    Employee e2 = {id:2, name:"B", salary:200};
+    Employee e3 = {id:3, name:"C", salary:300};
+
+    dt.add(e1);
+    dt.add(e2);
+    dt.add(e3);
+
+    return dt;
+}
+
+function testSelect() (table) {
+
+    table<Employee> dt =  createTable();
+
+    table<EmployeeSalary> salaryTable = dt.select(getSalary);
+    return salaryTable;
+}
+
+function testSelectCompatibleLambdaInput() (table) {
+    table<Employee> dt = createTable();
+
+    table<EmployeeSalary> salaryTable = dt.select(getSalaryCompatibleInput);
+    return salaryTable;
+}
+
+function testSelectCompatibleLambdaOutput() (table) {
+    table<Employee> dt = createTable();
+
+    table<EmployeeSalary> salaryTable = dt.select(getSalaryCompatibleOutput);
+    return salaryTable;
+}
+
+function testSelectCompatibleLambdaInputOutput() (table) {
+    table<Employee> dt = createTable();
+
+    table<EmployeeSalary> salaryTable = dt.select(getSalaryCompatibleInputOutput);
+    return salaryTable;
+}
+
```

Comment:
Shall we move these into table-iteration.bal

Reply:
Done